src/NLog/Config/ConfigurationItemFactory.cs

```diff
@@ -202,8 +204,8 @@ public IJsonConverter JsonConverter
         [Obsolete("Instead use LogFactory.ServiceRepository.ResolveInstance(typeof(IValueFormatter)). Marked obsolete on NLog 5.0")]
         public IValueFormatter ValueFormatter
         {
-            get => _serviceResolver.ResolveValueFormatter();
-            set => (_serviceResolver as IServiceRepository)?.RegisterValueFormatter(value);
+            get => _serviceResolver.ResolveService<IValueFormatter>() ?? MessageTemplates.ValueFormatter.Instance;
```

Comment:
If resolver is always right, then you can remove the fallback.

Reply:
good one, fixed